From 2a4149fc4219512485ae46b99349cac9f31ed5b9 Mon Sep 17 00:00:00 2001
From: "Yang, Dong" <dong.yang@intel.com>
Date: Thu, 29 Nov 2018 15:30:06 +0800
Subject: [PATCH 0871/1214] drm/i915: Remove __GFP_RETRY_MAYFAIL flag

The RenderThread crash when open DRM device failed, its due to
system memory exhaused.
Remove the __GFP_RETRY_MAYFAIL flag to block wait for memory if
system run out of memory.

Change-Id: I045e4d17987e8f4fe234247bc7f3b3ee5e0a9df0
Tracked-On: OAM-72153
Signed-off-by: Yang, Dong <dong.yang@intel.com>
---
 drivers/gpu/drm/i915/i915_gem_gtt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_gem_gtt.c b/drivers/gpu/drm/i915/i915_gem_gtt.c
index d6baa2d..772168d 100644
--- a/drivers/gpu/drm/i915/i915_gem_gtt.c
+++ b/drivers/gpu/drm/i915/i915_gem_gtt.c
@@ -649,7 +649,7 @@ setup_scratch_page(struct i915_address_space *vm, gfp_t gfp)
 		size = I915_GTT_PAGE_SIZE_64K;
 		gfp |= __GFP_NOWARN;
 	}
-	gfp |= __GFP_ZERO | __GFP_RETRY_MAYFAIL;
+	gfp |= __GFP_ZERO;
 
 	do {
 		int order = get_order(size);
-- 
2.7.4

